### PR TITLE
Fix28

### PR DESCRIPTION
**Pull request by avishmehta68710**
Added a verification for email address using regex

### CODE REVIEW

File: lib/constants/validators.dart [hunk 1]
```diff
@@ -54,3 +56,17 @@ Function(String) bloodGroupValidator = (String bloodGroup) {
 
   return null;
 };
+
+
+Function(String) emailAddressValidator = (String emailaddress) {
+  String email = r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]";
+  bool valids = RegExp(email).hasMatch(emailaddress);
```
Comment (arteevraina) [inline lib/constants/validators.dart:63]: rather than `valids` rename this variable as `isValid`

File: lib/constants/validators.dart [hunk 2]
```diff
@@ -54,3 +56,17 @@ Function(String) bloodGroupValidator = (String bloodGroup) {
...
+
+  if (emailaddress.isEmpty) {
+    return "Please enter a valid email";
+  }
+  if (valids == false) {
```
Comment (arteevraina) [inline lib/constants/validators.dart:68]: It can be now written as `!isValid`

File: lib/constants/validators.dart [hunk 3]
```diff
@@ -1,6 +1,8 @@
-// Functions for validations of usernames, passwords, phonenumbers, bloodgroups.
+// Functions for validations of usernames, passwords, phonenumbers, bloodgroups, emailaddresses.
 // More functions will be added as the app grows.
 
+// import 'dart:html';
```
Comment (arteevraina) [inline lib/constants/validators.dart:4]: Why added this line?

**Review by arteevraina — CHANGES_REQUESTED** [review]

**Comment by arteevraina** [comment]
@avishmehta68710 You need to read the [contributing guidelines](https://github.com/arteevraina/KnowYourDonor/blob/master/CONTRIBUTING.md) before making any PR. This means you have to write meaningful commit messages and update your PR with the well-defined description of what it does or simply write Fixes: #issueNumber. And last thing is that please don't push the files like pubspec.lock or other files(or binaries) that you have not changed.

File: lib/constants/validators.dart (hunk 3 above)
Comment (avishmehta68710) [reply lib/constants/validators.dart:4]: sorry that line was added y mistake will delete it

File: lib/constants/validators.dart (hunk 2 above)
Comment (avishmehta68710) [reply lib/constants/validators.dart:68]: if is write isValid == false. is it wrong ?????

Comment (arteevraina) [reply lib/constants/validators.dart:68]: No, it is not wrong. But it is not the best way to write this.

**Comment by arteevraina** [comment]
ref: #36

**Pull request closed without merging**